src/bsoncxx/builder/core.cpp

```diff
@@ -214,7 +214,7 @@ class core::impl {
             }
         }
 
-        ~frame() {
+        ~frame() noexcept(false) {
```

Comment:
I agree, but the ~frame already throws the exception. I don't think it should do that, but also am unsure whether we can "just remove it". Hence, for now I thought it'd be better to add the right ``noexcept()`` so that things at least compile

Reply:
I agree, but the dtor currently *does* throw, and by marking it as such, it at least does not thrown any warnings that prevent compilation with -Werror. I am happy to make a ticket to fix this, however, I am not sure how yet and not break the "API".

Reply:
Additionally, this seems independent of the goal of addressing deprecations in the associated commit. It should probably have its own ticket and commit.

Reply:
Originally that destructor didn't throw (it silently ate errors).  That was probably a mistake, but a minor one (because the only failure mode for array/doc_end is if you've exceeded the max buffer size).

The "correct" solution is probably to provide a separate close() instead of ~frame().  failure of close() could either throw or be acted upon to throw.  ~frame() would become a noop.

Reply:
I've created a ticket for this: https://jira.mongodb.org/browse/CXX-1458

Reply:
@hanumantmk - Wouldn't adding a `close` method defeat the purpose of frame, which is basically an RAII type? Should we just be only doing the throw if `std::uncaught_exception` returns false?